### PR TITLE
Yrob/11742 breadcrumbs

### PR DESCRIPTION
**Pull request by Yrob**
Breadcrumbs zijn nu:
`DSO-API Datasets/<dataset-naam>/<tabel-naam>/<instance-title>`
Of voor datasets op een sub path:
`DSO-API Datasets/<dataset-path>/<dataset-naam>/<tabel-naam>/<instance-title>`

Ook de header in browser view is geupdate. Daar staat nu niet meer "Django Rest Framework", maar "Amsterdam Datadiensten"
met het logo van Gemeente Amsterdam.

### CODE REVIEW

File: src/dso_api/dynamic_api/routers.py [hunk 1]
```diff
@@ -48,14 +48,21 @@
     from schematools.contrib.django.models import DynamicModel
 
 
-class DynamicAPIRootView(APIIndexView):
+class DynamicAPIIndexView(APIIndexView):
```
Comment (ngaranko) [inline src/dso_api/dynamic_api/routers.py:51]: There's no need to rename this class, it is still Root view.

File: src/rest_framework_dso/views.py [hunk 2]
```diff
@@ -20,6 +21,35 @@
...
+    This name is used in the browsable API, and in OPTIONS responses.
+
+    This function repaces the Rest Framework default.
+    """
+    name = getattr(view, "name", None)
+    if name is not None:
+        return name
+
+    name = view.__class__.__name__
+    name = formatting.remove_trailing_string(name, "View")
```
Comment (jjmurre) [inline src/rest_framework_dso/views.py:36]: Nice, didn't know these util functions.

File: src/dso_api/dynamic_api/routers.py [hunk 3]
```diff
@@ -250,6 +262,35 @@ def _build_openapi_views(self, datasets: Iterable[Dataset]) -> List[URLPattern]:
...
+            paths += ds_paths[0:-1]
+        paths = set(paths)
+
+        # Create an index view for each path
+        results = []
+        for p in paths:
+            name = p.split("/")[-1].capitalize() + " Datasets"
+            results.append(
+                path(
+                    p + "/",
```
Comment (ngaranko) [inline src/dso_api/dynamic_api/routers.py:287]: What if multiple datasets are using same subpath? How this will look?

**Review by ngaranko — COMMENTED** [review]
Please add tests for every change in this PR.

File: src/templates/dso_api/dynamic_api/api.html [hunk 4]
```diff
@@ -0,0 +1,70 @@
+{% extends "rest_framework/base.html" %}
```
Comment (jjmurre) [inline src/templates/dso_api/dynamic_api/api.html:1]: Brand building :-)

**Review by jjmurre — APPROVED** [review]
LGTM

File: src/dso_api/dynamic_api/routers.py (hunk 3 above)
Comment (Yrob) [reply src/dso_api/dynamic_api/routers.py:287]: Like the index page at root, but with only the datasets on that path

File: src/dso_api/dynamic_api/routers.py (hunk 1 above)
Comment (Yrob) [reply src/dso_api/dynamic_api/routers.py:51]: But its now also an index view at locations other than root

Comment (ngaranko) [reply src/dso_api/dynamic_api/routers.py:51]: Those are roots of subtrees, right? :)

Comment (Yrob) [reply src/dso_api/dynamic_api/routers.py:51]: So it should be called DynamicAPINodeView? or DynamicAPISubTreeView :-)

**Comment by Yrob** [comment]
> Please add tests for every change in this PR.

Done

File: src/dso_api/dynamic_api/routers.py (hunk 1 above)
Comment (ngaranko) [reply src/dso_api/dynamic_api/routers.py:51]: Why not keep DynamicAPIRootView? It's just fine.

File: src/dso_api/dynamic_api/routers.py (hunk 3 above)
Comment (ngaranko) [reply src/dso_api/dynamic_api/routers.py:287]: This is still not covered by tests.

Comment (Yrob) [reply src/dso_api/dynamic_api/routers.py:287]: Doesnt the new test in  src/tests/test_openapi.py cover this? It has two datasets on the same subpath and checks both are listed.

Comment (ngaranko) [reply src/dso_api/dynamic_api/routers.py:287]: Oh, it does, missed.

**Pull request merged**